File: DemoMethods/PythonDemo/demo_methods/Basic/StaticQuery.py
from demo.holder import DocumentStoreHolder
from datetime import timedelta
from pyravendb.tools.utils import Utils


class StaticQuery(object):
    @staticmethod
    def static_query(high_price=500, delay_days=35):
        with DocumentStoreHolder.get_store().open_session() as session:
            query = list(
                session.query(index_name="CostlyOrders").
                where_greater_than("Price", high_price).
                and_also().
                where_greater_than("Delay", Utils.timedelta_tick(timedelta(days=delay_days))))

            return query[0] if query else None

File: DemoServer/Demos/Basic/StaticQuery.cs
using System;
using System.Linq;
using DemoServer.Controllers;
using DemoServer.Entities;
using DemoServer.Helpers;
using DemoServer.Indexes;
using Microsoft.AspNetCore.Mvc;

namespace DemoServer.Demos.Basic
{
    public partial class BasicController : BaseController
    {
        [HttpGet]
        [Route("/basic/staticQuery")]
        [Demo("Static Query", DemoOutputType.Flatten, demoOrder: 70)]
        public object StaticQuery(string highPrice = "500", string delayDays = "35")
        {
            var HighPrice = int.Parse(highPrice);
            var DelayDays = int.Parse(delayDays);

            //GetImportantOrdersWithIssues - High Price Orders and Delayed Orders
            using (var session = DocumentStoreHolder.Store.OpenSession())
            {
                var problemtaticOrders = session
                    .Query<CostlyOrders.Result, CostlyOrders>()
                    .Where(x => x.Price > HighPrice && x.Delay > TimeSpan.FromDays(DelayDays))
                    .OfType<Order>()
                    .ToList();

                return problemtaticOrders;
            }
        }
    }
}
